src/lib/modules/ping/ping.go

```diff
@@ -21,6 +21,10 @@ func HandlePing(globalChain []*tsdb.ChainPing, urlRaw *string, packets int, tsdb
 	// launch a goroutine to handle ping operations
 	go utils.CLIPing(urlRaw, packets, chnl)
 	resp := <-chnl
+	if *resp == "" {
```

Comment:
if the `ping` works abnormally, then it should return `.... Name or service not known` rather than an empty string, since the golang subprocess module jumps to the next line of execution-only after finishing a particular instruction, and since the mechanism is not normal, it should return an exception than an empty string.
Please have a look.

Reply:
Sure. That would be good